testing/endtoend/evaluators/beaconapi_evaluators/beacon_api.go

```diff
@@ -89,121 +209,267 @@ var beaconPathsAndObjects = map[string]metadata{
-				return fmt.Errorf("attester data does not match, prysm: %d lighthouse: %d", len(castedp.Data), len(castedl.Data))
+			if pResp.Data == nil {
+				return errEmptyPrysmData
 			}
-			return compareJSONResponseObjects(prysmResp, castedl)
+			if lResp.Data == nil {
+				return errEmptyLighthouseData
+			}
+			return nil
 		},
 	},
-	"/beacon/headers/{param1}": {
-		basepath: v1MiddlewarePathTemplate,
-		params: func(_ string, e primitives.Epoch) []string {
-			slot := uint64(0)
-			if e > 0 {
-				slot = (uint64(e) * uint64(params.BeaconConfig().SlotsPerEpoch)) - 1
+	"/beacon/pool/voluntary_exits": {
+		basepath: v1PathTemplate,
+		params: func(_ string, _ primitives.Epoch) []string {
+			return []string{}
+		},
+		prysmResps: map[string]interface{}{
+			"json": &beacon.ListVoluntaryExitsResponse{},
+		},
+		lighthouseResps: map[string]interface{}{
+			"json": &beacon.ListVoluntaryExitsResponse{},
+		},
+		customEvaluation: func(p interface{}, l interface{}) error {
+			pResp, ok := p.(*beacon.ListVoluntaryExitsResponse)
+			if !ok {
+				return errJsonCast
 			}
-			return []string{fmt.Sprintf("%v", slot)}
+			lResp, ok := l.(*beacon.ListVoluntaryExitsResponse)
+			if !ok {
+				return errJsonCast
+			}
+			if pResp.Data == nil {
+				return errEmptyPrysmData
+			}
+			if lResp.Data == nil {
+				return errEmptyLighthouseData
+			}
+			return nil
+		},
+	},
+	"/beacon/pool/bls_to_execution_changes": {
+		basepath: v1PathTemplate,
+		params: func(_ string, _ primitives.Epoch) []string {
+			return []string{}
 		},
 		prysmResps: map[string]interface{}{
-			"json": &beacon.GetBlockHeaderResponse{},
+			"json": &beacon.BLSToExecutionChangesPoolResponse{},
 		},
 		lighthouseResps: map[string]interface{}{
-			"json": &beacon.GetBlockHeaderResponse{},
+			"json": &beacon.BLSToExecutionChangesPoolResponse{},
+		},
+		customEvaluation: func(p interface{}, l interface{}) error {
+			pResp, ok := p.(*beacon.BLSToExecutionChangesPoolResponse)
+			if !ok {
+				return errJsonCast
+			}
+			lResp, ok := l.(*beacon.BLSToExecutionChangesPoolResponse)
+			if !ok {
+				return errJsonCast
+			}
+			if pResp.Data == nil {
+				return errEmptyPrysmData
+			}
+			if lResp.Data == nil {
+				return errEmptyLighthouseData
+			}
+			return nil
 		},
 	},
-	// we want to test comma-separated query params
-	"/beacon/states/{param1}/validators?id=0,1": {
-		basepath: v1MiddlewarePathTemplate,
-		params: func(_ string, e primitives.Epoch) []string {
+	"/config/fork_schedule": {
+		basepath: v1PathTemplate,
+		params: func(_ string, _ primitives.Epoch) []string {
+			return []string{}
+		},
+		prysmResps: map[string]interface{}{
+			"json": &apimiddleware.ForkScheduleResponseJson{},
+		},
+		lighthouseResps: map[string]interface{}{
+			"json": &apimiddleware.ForkScheduleResponseJson{},
+		},
+		customEvaluation: func(p interface{}, l interface{}) error {
+			// remove all forks with far-future epoch
+			pSchedule, ok := p.(*apimiddleware.ForkScheduleResponseJson)
```

Comment:
do we only have an api middleware representation? maybe we aren't able to remove this yet?

Reply:
This endpoint has not been converted to HTTP yet